### DISCUSSION TITLE
課題12についての質問

### DISCUSSION THREAD

**Discussion by TakedaAyano** [nsc-kiwamu/traning-takeda#47] [Q&A]
問題４についてなのですが、
4人がサイコロを振って、出た目の数が一番大きい数だけを、maxに格納することはできたのですが、それ以降の「勝ち負け判定」が上手くいきません。
for文内にconsole.logを書くと何回も出てしまうので、{ }外に書くことはわかるのですが、その出力する際の条件式がわかりません。

これも、人数分の式を作成して出力するのではないかと思い実行してみたのですが、ただ全員分の名前が出るだけで、該当する人だけを出すことができませんでした。
どのようにすればよろしいでしょうか？

**Comment by nsc-kiwamu** [comment]
まずは勝ち負け判定のfor文の条件が正しくないです。
for (let user = 0; user < 3; user++) { // 4人分繰り返す
↑4人分繰り返すと記載していますが、3人分の繰り返しですね。
for (let user = 0; user < 4; user++) { 
としてもまぁ、良いですが
「user < 4」のところはもっと適切な記載があるはずです。
※勝負する人数は配列で記載しているので、その長さを使う

if (users4[0].matchValue[0] == max){ // 最大値と同じであれば勝ち
これだと常に1番目の人の比較ですよね？
ループさせる毎に1番目～4番目の人の結果とmax値を比較したい
